tests/integration/solidity_contracts/PlainOpcodes/test_plain_opcodes.py

```diff
@@ -71,242 +67,14 @@ async def test_should_return_self_address(
-        )
-        async def test_extcodecopy_offset_and_size_gte_twenty_c(self, counter, kakarot):
-            evm_contract_address = (
-                counter.contract_account.deploy_call_info.result.evm_contract_address
-            )
-            expected_memory_result = hex_string_to_bytes_array(counter.bytecode.hex())
-
-            # instructions
-            push1 = 60
-            push20 = 73
-            extcodecopy = "3c"
-
-            # stack elements
-            offset = 18
-            size = 3
-            dest_offset = 0
-
-            byte_code = f"{push1}\
-            {size:02x}\
-            {push1}\
-            {offset:02x}\
-            {push1}\
-            {dest_offset:02x}\
-            {push20}\
-            {evm_contract_address:x}\
-            {extcodecopy}"
-
-            res = await kakarot.execute(
-                value=int(0),
-                bytecode=hex_string_to_bytes_array(byte_code),
-                calldata=hex_string_to_bytes_array(""),
-            ).call(caller_address=1)
-
-            memory_result = extract_memory_from_execute(res.result)
-
-            assert (
-                memory_result[dest_offset : size + dest_offset]
-                == expected_memory_result[offset : offset + size]
-            )
-
-        @pytest.mark.skip(
-            "Investigate why there is a difference between local and deployed contract code in the erc20 contract"
-        )
-        async def test_extcodecopy_erc20(self, erc_20, kakarot):
-            evm_contract_address = (
-                erc_20.contract_account.deploy_call_info.result.evm_contract_address
-            )
-
-            # instructions
-            push1 = 60
-            push20 = 73
-            extcodecopy = "3c"
-
-            # stack elements
-            offset = 0
-            size = 8
-            dest_offset = 0
-
-            byte_code = f"{push1}\
-            {size:02x}\
-            {push1}\
-            {offset:02x}\
-            {push1}\
-            {dest_offset:02x}\
-            {push20}\
-            {evm_contract_address:x}\
-            {extcodecopy}"
-
-            res = await kakarot.execute(
-                value=int(0),
-                bytecode=hex_string_to_bytes_array(byte_code),
-                calldata=hex_string_to_bytes_array(""),
-            ).call(caller_address=1)
-
-            expected_memory_result = hex_string_to_bytes_array(erc_20.bytecode.hex())
-            memory_result = extract_memory_from_execute(res.result)
-            # asserting to the first discrepancy
-            assert memory_result[dest_offset:2] == expected_memory_result[offset:2]
+        @pytest.mark.parametrize("offset, size", [[0, 32], [32, 32], [0, None]])
+        async def test_should_return_counter_code(
+            self, plain_opcodes, counter, offset, size
+        ):
+            deployed_bytecode = counter.bytecode[counter.bytecode.index(0xFE) + 1 :]
```

Comment:
Can you leave a comment here explaining how this range was derived ?

I asked monsieur chat gpt and it filled me in:

> The reason for this is that the "0xFE" hex value is used as a marker for the beginning of the deployed bytecode. This is because the bytecode that is stored on the blockchain is different from the bytecode of the contract's source code. The bytecode stored on the blockchain includes additional information like the contract's deployed address and any libraries that it references.
> 
> So this python code is looking for the first occurrence of "0xFE" in the bytecode, which marks the beginning of the deployed bytecode, and then slicing the bytecode to include everything after that point. This ensures that the "deployed_bytecode" variable only includes the actual bytecode that is deployed on the blockchain, and not any additional information.
> 
> 

This was sort of the context I was missing for the #342 issue, right?

Reply:
Actually when you deploy a contract:
- you send some bytecode to address 0
- this bytecode is run as any bytecode
- the returned data of this run is the one deployed

And I thinks that it is only a convention (maybe depending on the lib?) to have the bytecode as follows:

```
   constructor bytecode                          contract bytecode                                 calldata
|------------------------FE|-------------------------------------------|--------------------|
```

the constructor bytecode is indeed not stored in the contract as it would be useless (constructor is only called at deploy time).

In other words, when you deploy a contract, you only run "constructor bytecode" (up to FE), and the output of this run is "contract bytecode" + possible initialization (Counter as no constructor indeed), which is stored